nearcore/src/runtime/mod.rs

```diff
@@ -796,20 +799,29 @@ impl RuntimeAdapter for NightshadeRuntime {
         gas_limit: Gas,
         challenges: &ChallengesResult,
         random_seed: CryptoHash,
-        generate_storage_proof: bool,
         is_new_chunk: bool,
         is_first_block_with_chunk_of_version: bool,
-        states_to_patch: SandboxStatePatch,
-        use_flat_storage: bool,
     ) -> Result<ApplyTransactionResult, Error> {
-        let trie =
-            self.get_trie_for_shard(shard_id, prev_block_hash, *state_root, use_flat_storage)?;
-
-        // TODO (#6316): support chunk nodes caching for TrieRecordingStorage
-        if generate_storage_proof {
-            panic!("Storage proof generation is not enabled yet");
+        let _timer =
+            metrics::APPLYING_CHUNKS_TIME.with_label_values(&[&shard_id.to_string()]).start_timer();
+
+        let mut trie = match storage_config.source {
+            StorageDataSource::Db => self.get_trie_for_shard(
+                shard_id,
+                prev_block_hash,
+                storage_config.state_root,
+                storage_config.use_flat_storage,
+            )?,
+            StorageDataSource::Recorded(storage) => Trie::from_recorded_storage(
+                storage,
+                storage_config.state_root,
+                storage_config.use_flat_storage,
+            ),
+        };
+        if storage_config.record_storage {
+            trie = trie.recording_reads()
```

Comment:
nit: semicolon (not worth getting a reapproval on, just merge).

Reply:
reapprovals should no longer be a thing, so I will fix that: [source](https://near.zulipchat.com/#narrow/stream/295302-general/topic/Github.20nearcore.20branch.20protection/near/398652191)